src/SubscribeBlockTracker.test.ts

```diff
@@ -0,0 +1,3221 @@
+import EMPTY_FUNCTION from '../tests/emptyFunction';
+import recordCallsToSetTimeout from '../tests/recordCallsToSetTimeout';
+import { withSubscribeBlockTracker } from '../tests/withBlockTracker';
+import buildDeferred from '../tests/buildDeferred';
+import { SubscribeBlockTracker } from '.';
+
+interface Sync {
+  oldBlock: string;
+  newBlock: string;
+}
+
+const METHODS_TO_ADD_LISTENER = ['on', 'addListener'] as const;
+const METHODS_TO_REMOVE_LISTENER = ['off', 'removeListener'] as const;
+const METHODS_TO_GET_LATEST_BLOCK = [
+  'getLatestBlock',
+  'checkForLatestBlock',
+] as const;
+
+describe('SubscribeBlockTracker', () => {
+  describe('constructor', () => {
+    it('should throw if given no options', () => {
+      expect(() => new SubscribeBlockTracker()).toThrow(
+        'SubscribeBlockTracker - no provider specified.',
+      );
+    });
+
+    it('should throw if given options but not given a provider', () => {
+      expect(() => new SubscribeBlockTracker({})).toThrow(
+        'SubscribeBlockTracker - no provider specified.',
+      );
+    });
+
+    it('should return a block tracker that is not running', async () => {
+      recordCallsToSetTimeout();
+
+      await withSubscribeBlockTracker(({ blockTracker }) => {
+        expect(blockTracker.isRunning()).toBe(false);
+      });
+    });
+  });
+
+  METHODS_TO_GET_LATEST_BLOCK.forEach((methodToGetLatestBlock) => {
+    describe(`${methodToGetLatestBlock}`, () => {
+      it('should start the block tracker immediately after being called', async () => {
+        recordCallsToSetTimeout();
+
+        await withSubscribeBlockTracker(async ({ blockTracker }) => {
+          const promiseToGetLatestBlock =
+            blockTracker[methodToGetLatestBlock]();
+          expect(blockTracker.isRunning()).toBe(true);
+          // We have to wait for the promise to resolve after the assertion
+          // because by the time this promise resolves, the block tracker isn't
+          // running anymore
+          await promiseToGetLatestBlock;
+        });
+      });
+
+      it('should stop the block tracker automatically after its promise is fulfilled', async () => {
+        recordCallsToSetTimeout();
+
+        await withSubscribeBlockTracker(async ({ blockTracker }) => {
+          await blockTracker[methodToGetLatestBlock]();
+          expect(blockTracker.isRunning()).toBe(false);
+        });
+      });
+
+      it('should fetch the latest block number', async () => {
+        recordCallsToSetTimeout();
+
+        await withSubscribeBlockTracker(
+          {
+            provider: {
+              stubs: [
+                {
+                  methodName: 'eth_blockNumber',
+                  response: {
+                    result: '0x0',
+                  },
+                },
+              ],
+            },
+          },
+          async ({ blockTracker }) => {
+            const latestBlockNumber = await blockTracker.getLatestBlock();
```

Comment:
Was this meant to use `methodToGetLatestBlock`?

```suggestion
            const latestBlockNumber = await blockTracker[methodToGetLatestBlock]();
```

Reply:
Yes, thank you!